trending_data/src/main/webapp/unique_window_handler.js

```diff
@@ -41,22 +31,24 @@ export class UniqueWindowHandler {
   }
 
   /**
-  * Called when Twitter button is clicked. Clears
+  * Called when YouTube button is clicked. Clears
   * current content of window and shows the most recently
-  * cached Twitter data
+  * cached YouTube data
   */
   showYTData() {
+    this.showing = 'yt';
     this.initDataWindow(); // clear current content and re-add buttons
     this.dataWindow.appendChild(this.ytDataDiv);
     this.currentWindow.setContent(this.dataWindow);
   }
 
   /**
-  * Called when YouTube button is clicked. Clears
+  * Called when Twitter button is clicked. Clears
   * current content of window and shows the most recently
-  * cached YouTube data
+  * cached Twitter data
   */
   showTwitterData() {
+    this.showing = 'twitter';
     this.initDataWindow();
     this.dataWindow.appendChild(this.twitterDataDiv);
     this.currentWindow.setContent(this.dataWindow);
```

Comment:
If someone clicks the button while already showing youtube. It unnecessarily loads the page again.

Reply:
See `openWindow()` - these methods are not even called if the user clicks on the same marker (there is a `return` statement)